scripts/configurator/parser/cmake_var.py

```diff
@@ -0,0 +1,39 @@
+def generate_cmake_var(type :str, var):
+   "factory function that associates the correct type of variable"
+   string_to_variable_class = {
+      "BOOL" : cmake_var_bool,
+      "INTERNAL" : cmake_var,
+      "FILEPATH": cmake_var,
+      "STRING" : cmake_var,
+   }
+   variable_instance = string_to_variable_class[type]
```

Comment:
NIT: No need for a variable (too long), but also, `variable_instance` is wrong: you're getting a `class` object, not an instance. So, at most it should be something like `class_obj`. Later, you're creating an object instance from the class.